### PR TITLE
build: update devenv

### PR DESCRIPTION
**Pull request by janniks**
- Adds CORS proxy to bitcoin service (via nginx). Useful for web-apps to use RPC methods (likely not a prod use-case, but great for dev/debugging) **NEW PORT ~18445~18433**
- ~Adds healthcheck to `stacks-api` → once docker-compose up completes, we can be sure stacks-api is ready for deploying contracts.~
  > ~lmk if I should remove this or add as separate PR~
- ~Some minor ARG changes for faster initial stacks mining/catchup (takes 45s on my machine)~
  > ~happy to tweak if other have better values~
- Changes `electrs` to `mempool/electrs` which includes a built-in HTTP API (Blockstream fork). This one has endpoints like `/utxo` **NEW PORT 3002**

### CODE REVIEW

File: devenv/miner/docker/entrypoint.sh
Comment (friedger) [inline devenv/miner/docker/entrypoint.sh:26]: Why 2 seconds for the first 20 blocks?

**Review by friedger — COMMENTED** [review]
Looks good. Could you add some details about the new feature in the devenv README?

File: devenv/electrs/docker/Dockerfile
Comment (AshtonStephens) [inline devenv/electrs/docker/Dockerfile:14]: Is there a chance that this breaks the other services that use electrs?

File: devenv/miner/docker/entrypoint.sh
Comment (CAGS295) [inline devenv/miner/docker/entrypoint.sh:24]: Why not generate 20 blocks in a single call?

File: devenv/docker-compose.yml
Comment (CAGS295) [inline devenv/docker-compose.yml:13]: I'd like to point out the resemblance with the mempool-api service. Are these two roughly the same?

Comment (AshtonStephens) [reply devenv/docker-compose.yml:13]: mempool is here because it's required by the leather wallet, but yes it's really our way of including an esplora-esque interface.

File: devenv/electrs/docker/Dockerfile
Comment (janniks) [reply devenv/electrs/docker/Dockerfile:14]: Might be, but I haven't seen any yet. I think it's still a fork and 90% compatible -- but I can't say for certain

File: devenv/miner/docker/entrypoint.sh
Comment (janniks) [reply devenv/miner/docker/entrypoint.sh:26]: This helped my network bootstrap faster (Stacks catch up), but I'll remove as unrelated to get the important fixes merged faster

File: devenv/miner/docker/entrypoint.sh
Comment (janniks) [reply devenv/miner/docker/entrypoint.sh:24]: This helped my network bootstrap faster (Stacks catch up), but I'll remove as unrelated to get the important fixes merged faster

**Comment by janniks** [comment]
Adding another change, seems preflight wasn't working on Chromium browsers.

**Review by friedger — DISMISSED** [review]
Could you update the README?

**Comment by friedger** [comment]
@kenrogers suggests that the healthcheck is not working properly

**Comment by janniks** [comment]
Oops sorry, I committed the healthchecks again by accident, will remove again. My bad

**Comment by AshtonStephens** [comment]
I'm running this right now @janniks - the stacks-api always times out and fails to become healthy. I think we should remove the `stacks-api` `service_healthy` check.

**Comment by janniks** [comment]
Yes, sorry. I didn't have access to my computer today. Will update tmrw

**Review by friedger — APPROVED** [review]
LGTM

**Review by AshtonStephens — APPROVED** [review]

**Comment by AshtonStephens** [comment]
@janniks  I'll let you merge. Looks good.

**Comment by janniks** [comment]
I don't have write access 🙈

**Comment by janniks** [comment]
Feel free to squash-merge away my intermediate commits

**Pull request merged**
